src/datadog/datadog_agent.cpp

```diff
@@ -433,10 +426,11 @@ void DatadogAgent::get_and_apply_remote_configuration_updates() {
         }
 
         if (!response_json.empty()) {
-          // TODO (during Active Configuration): `process_response` should
-          // return a list of configuration update and should be consumed by
-          // telemetry.
-          remote_config_.process_response(response_json);
+          auto updated_configuration =
+              remote_config_.process_response(response_json);
+          if (!updated_configuration.empty()) {
+            send_configuration_change(updated_configuration);
+          }
```

Comment:
You might run into a bug that I fixed in an unmerged [ancient branch][1].

We're currently in a callback (response handler) to `class Curl`. `Curl`'s mutex is locked, so when we try to enqueue a HTTP request inside `send_configuration_change`, I think we'll have a deadlock.

Easy enough to fix `class Curl`.

[1]: https://github.com/DataDog/dd-trace-cpp/compare/main...david.goffredo/curl-event-loop#diff-a747ca63fc60398d29a5669c3a753950c18e737d84cbd518fca99f5752bd60abR443-R458

Reply:
Nope, we fixed it a while back.